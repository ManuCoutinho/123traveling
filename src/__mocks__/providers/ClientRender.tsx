/* eslint-disable react/display-name */

import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import { ChakraProvider } from '@chakra-ui/react'

const createTestQueryClient = () =>
	new QueryClient({
		defaultOptions: {
			queries: {
				retry: false
			}
		}
	})

export function renderWithClient(children: React.ReactNode) {
	const testQueryClient = createTestQueryClient()
	const { rerender, ...result } = render(
		<ChakraProvider>
			<QueryClientProvider client={testQueryClient}>
				{children}
			</QueryClientProvider>
		</ChakraProvider>
	)
	return {
		...result,
		rerender: (rerenderUi: React.ReactNode) =>
			rerender(
				<QueryClientProvider client={testQueryClient}>
					{rerenderUi}
				</QueryClientProvider>
			)
	}
}

export const customWrapper = () => {
	const testQueryClient = createTestQueryClient()

	return ({ children }: { children: React.ReactNode }) => (
		<QueryClientProvider client={testQueryClient}>
			{children}
		</QueryClientProvider>
	)
}
