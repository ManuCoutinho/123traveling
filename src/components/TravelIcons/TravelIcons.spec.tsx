import { screen } from '@testing-library/react'
import { TravelIcons } from '.'
import { renderWithClient } from '__mocks__/providers/ClientRender'

describe('<TravelIcons/>', () => {
	it('should render TravelIcons correctly ', () => {
		renderWithClient(<TravelIcons />)
		expect(
			screen.getByRole('button', { name: /moderno/i })
		).toBeInTheDocument()
		expect(
			screen.getByRole('button', { name: /vida noturna/i })
		).toBeInTheDocument()
		expect(
			screen.getByRole('button', { name: /clÃ¡ssico/i })
		).toBeInTheDocument()

		expect(
			screen.getByRole('button', { name: /praia/i })
		).toBeInTheDocument()
	})
	it('should match to snapshot', () => {
		const { container } = renderWithClient(<TravelIcons />)
		expect(container).toMatchSnapshot()
	})
})
