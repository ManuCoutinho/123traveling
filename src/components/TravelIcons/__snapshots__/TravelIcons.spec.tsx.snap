// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TravelIcons/> should match to snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="chakra-wrap css-ktuo2q"
    >
      <ul
        class="chakra-wrap__list css-148behr"
      >
        <li
          class="css-12ncwpk"
        >
          <button
            aria-label="vida noturna"
            class="chakra-button css-dz9mkr"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1.25em"
              stroke="currentColor"
              stroke-width="0"
              style="vertical-align: middle;"
              viewBox="0 0 576 512"
              width="1.25em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M296 464h-56V338.78l168.74-168.73c15.52-15.52 4.53-42.05-17.42-42.05H24.68c-21.95 0-32.94 26.53-17.42 42.05L176 338.78V464h-56c-22.09 0-40 17.91-40 40 0 4.42 3.58 8 8 8h240c4.42 0 8-3.58 8-8 0-22.09-17.91-40-40-40zM432 0c-62.61 0-115.35 40.2-135.18 96h52.54c16.65-28.55 47.27-48 82.64-48 52.93 0 96 43.06 96 96s-43.07 96-96 96c-14.04 0-27.29-3.2-39.32-8.64l-35.26 35.26C379.23 279.92 404.59 288 432 288c79.53 0 144-64.47 144-144S511.53 0 432 0z"
              />
            </svg>
          </button>
          <p
            class="chakra-text css-1srqeis"
          >
            vida noturna
          </p>
        </li>
        <li
          class="css-12ncwpk"
        >
          <button
            aria-label="praia"
            class="chakra-button css-dz9mkr"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1.25em"
              stroke="currentColor"
              stroke-width="0"
              style="vertical-align: middle;"
              viewBox="0 0 640 512"
              width="1.25em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M115.38 136.9l102.11 37.18c35.19-81.54 86.21-144.29 139-173.7-95.88-4.89-188.78 36.96-248.53 111.8-6.69 8.4-2.66 21.05 7.42 24.72zm132.25 48.16l238.48 86.83c35.76-121.38 18.7-231.66-42.63-253.98-7.4-2.7-15.13-4-23.09-4-58.02.01-128.27 69.17-172.76 171.15zM521.48 60.5c6.22 16.3 10.83 34.6 13.2 55.19 5.74 49.89-1.42 108.23-18.95 166.98l102.62 37.36c10.09 3.67 21.31-3.43 21.57-14.17 2.32-95.69-41.91-187.44-118.44-245.36zM560 447.98H321.06L386 269.5l-60.14-21.9-72.9 200.37H16c-8.84 0-16 7.16-16 16.01v32.01C0 504.83 7.16 512 16 512h544c8.84 0 16-7.17 16-16.01v-32.01c0-8.84-7.16-16-16-16z"
              />
            </svg>
          </button>
          <p
            class="chakra-text css-1srqeis"
          >
            praia
          </p>
        </li>
        <li
          class="css-12ncwpk"
        >
          <button
            aria-label="moderno"
            class="chakra-button css-dz9mkr"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1.25em"
              stroke="currentColor"
              stroke-width="0"
              style="vertical-align: middle;"
              viewBox="0 0 16 16"
              width="1.25em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 2.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1Zm3 0a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1Zm3.5-.5a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1ZM4 5.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1ZM7.5 5a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1Zm2.5.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1ZM4.5 8a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1Zm2.5.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1Zm3.5-.5a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1Z"
              />
              <path
                d="M2 1a1 1 0 0 1 1-1h10a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V1Zm11 0H3v14h3v-2.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5V15h3V1Z"
              />
            </svg>
          </button>
          <p
            class="chakra-text css-1srqeis"
          >
            moderno
          </p>
        </li>
        <li
          class="css-12ncwpk"
        >
          <button
            aria-label="clássico"
            class="chakra-button css-dz9mkr"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1.25em"
              stroke="currentColor"
              stroke-width="0"
              style="vertical-align: middle;"
              viewBox="0 0 16 16"
              width="1.25em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m8 0 6.61 3h.89a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5H15v7a.5.5 0 0 1 .485.38l.5 2a.498.498 0 0 1-.485.62H.5a.498.498 0 0 1-.485-.62l.5-2A.501.501 0 0 1 1 13V6H.5a.5.5 0 0 1-.5-.5v-2A.5.5 0 0 1 .5 3h.89L8 0ZM3.777 3h8.447L8 1 3.777 3ZM2 6v7h1V6H2Zm2 0v7h2.5V6H4Zm3.5 0v7h1V6h-1Zm2 0v7H12V6H9.5ZM13 6v7h1V6h-1Zm2-1V4H1v1h14Zm-.39 9H1.39l-.25 1h13.72l-.25-1Z"
              />
            </svg>
          </button>
          <p
            class="chakra-text css-1srqeis"
          >
            clássico
          </p>
        </li>
        <li
          class="css-12ncwpk"
        >
          <button
            aria-label="e mais..."
            class="chakra-button css-dz9mkr"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1.25em"
              stroke="currentColor"
              stroke-width="0"
              style="vertical-align: middle;"
              viewBox="0 0 496 512"
              width="1.25em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm160 215.5v6.93c0 5.87-3.32 11.24-8.57 13.86l-15.39 7.7a15.485 15.485 0 0 1-15.53-.97l-18.21-12.14a15.52 15.52 0 0 0-13.5-1.81l-2.65.88c-9.7 3.23-13.66 14.79-7.99 23.3l13.24 19.86c2.87 4.31 7.71 6.9 12.89 6.9h8.21c8.56 0 15.5 6.94 15.5 15.5v11.34c0 3.35-1.09 6.62-3.1 9.3l-18.74 24.98c-1.42 1.9-2.39 4.1-2.83 6.43l-4.3 22.83c-.62 3.29-2.29 6.29-4.76 8.56a159.608 159.608 0 0 0-25 29.16l-13.03 19.55a27.756 27.756 0 0 1-23.09 12.36c-10.51 0-20.12-5.94-24.82-15.34a78.902 78.902 0 0 1-8.33-35.29V367.5c0-8.56-6.94-15.5-15.5-15.5h-25.88c-14.49 0-28.38-5.76-38.63-16a54.659 54.659 0 0 1-16-38.63v-14.06c0-17.19 8.1-33.38 21.85-43.7l27.58-20.69a54.663 54.663 0 0 1 32.78-10.93h.89c8.48 0 16.85 1.97 24.43 5.77l14.72 7.36c3.68 1.84 7.93 2.14 11.83.84l47.31-15.77c6.33-2.11 10.6-8.03 10.6-14.7 0-8.56-6.94-15.5-15.5-15.5h-10.09c-4.11 0-8.05-1.63-10.96-4.54l-6.92-6.92a15.493 15.493 0 0 0-10.96-4.54H199.5c-8.56 0-15.5-6.94-15.5-15.5v-4.4c0-7.11 4.84-13.31 11.74-15.04l14.45-3.61c3.74-.94 7-3.23 9.14-6.44l8.08-12.11c2.87-4.31 7.71-6.9 12.89-6.9h24.21c8.56 0 15.5-6.94 15.5-15.5v-21.7C359.23 71.63 422.86 131.02 441.93 208H423.5c-8.56 0-15.5 6.94-15.5 15.5z"
              />
            </svg>
          </button>
          <p
            class="chakra-text css-1srqeis"
          >
            e mais...
          </p>
        </li>
      </ul>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
